src/base/net/downloadmanager.cpp

```diff
@@ -104,11 +102,10 @@ class Net::DownloadManager::NetworkCookieJar final : public QNetworkCookieJar
     {
         const QDateTime now = QDateTime::currentDateTime();
         QList<QNetworkCookie> cookies = cookieList;
-        for (const QNetworkCookie &cookie : cookieList)
+        Algorithm::removeIf(cookies, [&now](const QNetworkCookie &cookie)
```

Comment:
I would also extract this code into separate function:
```c++
void removeOutdatedCookies(QList<QNetworkCookie> &cookies)
{
    Algorithm::removeIf(cookies, [&now](const QNetworkCookie &cookie)
    {
        return !cookie.isSessionCookie() && (cookie.expirationDate() <= now);
    });
}
```

Reply:
Maybe not in this PR, there are 2 kinds of predicate in use.

Reply:
> there are 2 kinds of predicate in use.

I misread it.


> Maybe not in this PR

Ok.